lib/msf/core/db_manager/host.rb

```diff
@@ -268,6 +269,16 @@ def report_host(opts)
   }
   end
 
+  def update_host(opts)
+    # TODO: remove unneeded workspace from opts until it's not automatically added to remote requests
+    opts.delete(:workspace)
```

Comment:
why is this done? - your todo isn't clear to me

Reply:
The `make_request` method in `http/core.rb` append a `workspace` key-value to every JSON request body. This is something that @jbarnett-r7 and I agree needs to be refactored, but it is out of scope for the current sprint tasks.

Reply:
So you didn't really answer my question which I will rephrase as such: What is he purpose of deleting the ":workspace" from the "opts" hash?

But I gather from your response the answer is there is really no purpose.  This worries me some because now I'm wondering which workspace the updating host change is associated with?

Reply:
It's a protection against the user accidentally updating the workspace. It should not be possible for workspaces to change on a host (or any DB object really) based on current functionality. Maybe it's something we want to add in the future?

Reply:
@clee-r7 Good point, I was in my head a bit when I quickly typed up the above response. The purpose of removing `:workspace` from `opts` is centered in the fact that it is unnecessarily added to the options for every request and in this situation `opts` is being passed directly into the `update` method. I believe the host identifier is all we should need for the update operation and the workspace should not be need to update a given host.